### PR TITLE
Solution implemented

### PR DESCRIPTION
**Pull request by NikolayAbysov**

### CODE REVIEW

File: src/main/java/core/basesyntax/FileWork.java [hunk 1]
```diff
@@ -9,6 +16,20 @@
...
-        return null;
+        String[] arr = new String[0];
+        try {
+            arr = Files.readString(Path.of(fileName)).toLowerCase().split("\\W+");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        List<String> list = new ArrayList<>();
+        for (String s : arr) {
+            if (s.indexOf('w') == 0) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:27]: create a constant for 'w'

File: src/main/java/core/basesyntax/FileWork.java [hunk 2]
```diff
@@ -9,6 +16,20 @@
  */
 public class FileWork {
     public String[] readFromFile(String fileName) {
-        return null;
+        String[] arr = new String[0];
+        try {
+            arr = Files.readString(Path.of(fileName)).toLowerCase().split("\\W+");
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:23]: don't ignore exception you should re-throw it

File: src/main/java/core/basesyntax/FileWork.java [hunk 3]
```diff
@@ -9,6 +16,20 @@
...
+        }
+        List<String> list = new ArrayList<>();
+        for (String s : arr) {
+            if (s.indexOf('w') == 0) {
+                list.add(s);
+            }
+        }
+        Collections.sort(list);
+        String[] output = new String[list.size()];
+        return list.toArray(output);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:33]:
```suggestion
        return list.toArray(String[]::new);
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 4]
```diff
@@ -8,7 +15,23 @@
  * Результат: web wide width world</p>
  */
 public class FileWork {
+
+    private static final char CHAR_MASK = 'w';
+
     public String[] readFromFile(String fileName) {
-        return null;
+        String[] arr;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:22]: let's give it more informative name

File: src/main/java/core/basesyntax/FileWork.java [hunk 5]
```diff
@@ -8,7 +15,23 @@
...
+        try {
+            arr = Files.readString(Path.of(fileName)).toLowerCase().split("\\W+");
+        } catch (IOException e) {
+            throw new RuntimeException("No file");
+        }
+        List<String> list = new ArrayList<>();
+        for (String s : arr) {
+            if (s.indexOf(CHAR_MASK) == 0) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:30]: let's make CHAR_MASK String and use string.startsWith() method

File: src/test/java/core/basesyntax/FileWorkTest.java [hunk 6]
```diff
@@ -3,6 +3,8 @@
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.io.IOException;
```
Comment (dmytro-hryhoruk) [inline src/test/java/core/basesyntax/FileWorkTest.java:6]: don't make any changes in tests

Comment (NikolayAbysov) [reply src/test/java/core/basesyntax/FileWorkTest.java]: Sorry

File: src/main/java/core/basesyntax/FileWork.java [hunk 7]
```diff
@@ -8,7 +15,23 @@
...
+
+    private static final String CHAR_MASK = "w";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        String[] words;
+        try {
+            words = Files.readString(Path.of(fileName)).toLowerCase().split("\\W+");
+        } catch (IOException e) {
+            throw new RuntimeException("No file");
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:26]: let's pass exception object that we have catched to RuntimeException constructor

**Pull request closed without merging**